extensions/amp-dailymotion/dailymotion-api.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright 2021 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {addParamsToUrl} from '../../src/url';
+import {dict} from '#core/types/object';
+import {isAutoplaySupported} from '#core/dom/video';
+
+/**
+ *
+ * @param win
+ * @param {string} videoId
+ * @param {string} endscreenEnable
+ * @param {string} info
+ * @param {string} mute
+ * @param {string} sharingEnable
+ * @param {string} start
+ * @param {string} uiHighlight
+ * @param {string} uiLogo
+ * @param {boolean} autoplay
+ * @param {!JsonObject} implicitParams
+ * @return {string}
+ */
+export function getDailymotionIframeSrc(
+  win,
+  videoId,
+  autoplay,
+  endscreenEnable,
+  info,
+  mute,
+  sharingEnable,
+  start,
+  uiHighlight,
+  uiLogo,
+  implicitParams
+) {
+  return addParamsToUrl(
+    `https://www.dailymotion.com/embed/video/${encodeURIComponent(
+      videoId
+    )}?api=1&html=1&app=amp`,
+    Object.assign(
+      dict({
+        'endscreen-enable': endscreenEnable,
+        'info': info,
+        // In order to support autoplay the video needs to be muted on load so we
+        // dont receive an unmute event which prevents the video from autoplay.
+        'mute':
+          mute || (autoplay && isAutoplaySupported(win)) ? true : undefined,
```

Comment:
I think we want to reverse the logic here because it should always be muted if autoplay is enabled, even if `data-mute` gives a different value via the API. We see this because in 0.1 the original `mute` param value gets overwritten under the `autoplay && isAutoplaySupported` conditions.
```suggestion
         autoplay ? 1 : mute,
```

Reply:
Whoops, you're right. 😆  Thank you!